### PR TITLE
Multiple Public Keys per Private Key

### PR DESCRIPTION
**Pull request by selfissued**
Added security considerations per Neil Madden's suggestions.

### CODE REVIEW

File: draft-ietf-oauth-jwk-thumbprint-uri.xml [hunk 1]
```diff
@@ -129,6 +129,32 @@
...
+
+      <section anchor="MultiplePublicKeysPerPrivateKey" title="Multiple Public Keys per Private Key">
+	<t>
+	  There are cryptographic algorithms for which multiple public keys correspond to the same private key.
+	  This is described in the security considerations of <xref target="RFC7748"/> as follows:
+	</t>
+	<t>
+	  <list style="empty">
+	    <t>
+	      Designers using these curves should be aware that for each public
```
Comment (Sakurann) [inline draft-ietf-oauth-jwk-thumbprint-uri.xml:141]: I think this should say "per each private key"?

File: draft-ietf-oauth-jwk-thumbprint-uri.xml [hunk 2]
```diff
@@ -129,6 +129,32 @@
...
+	      key, there are several publicly computable public keys that are
+	      equivalent to it, i.e., they produce the same shared secrets.  Thus
+	      using a public key as an identifier and knowledge of a shared secret
+	      as proof of ownership (without including the public keys in the key
+	      derivation) might lead to subtle vulnerabilities.
```
Comment (Sakurann) [inline draft-ietf-oauth-jwk-thumbprint-uri.xml:146]: I think "key derivation" should be clarified a little more.

File: draft-ietf-oauth-jwk-thumbprint-uri.xml [hunk 3]
```diff
@@ -129,6 +129,32 @@
...
+	    </t>
+	  </list>
+	</t>
+	<t>
+	  This applies to JWK Thumbprint URIs as identifiers equally as to public keys as identifiers.
```
Comment (Sakurann) [inline draft-ietf-oauth-jwk-thumbprint-uri.xml:151]: Could we paraphrase this sentence to make it a little bit clearer? Suggest:
> This consideration for public keys as identifiers equally applies to JWK Thumbprint URIs used as identifiers.

File: draft-ietf-oauth-jwk-thumbprint-uri.xml (hunk 1 above)
Comment (selfissued) [reply draft-ietf-oauth-jwk-thumbprint-uri.xml:141]: I believe the authors are saying that other equivalent public keys can be computed from a given public key.

Besides, this is a quotation from another RFC so we can't change it.

File: draft-ietf-oauth-jwk-thumbprint-uri.xml (hunk 2 above)
Comment (selfissued) [reply draft-ietf-oauth-jwk-thumbprint-uri.xml:146]: I actually think that the RFC authors misspoke by using the term "key derivation" above.  I believe they are actually talking about including the public key in the derivation or transmission of the identifier used.  That said, I could be wrong and if so, I hope Neil will point out how the current security considerations text should be improved.

We're accomplishing this as described in the following paragraph "A recommended way to ensure that the JWK Thumbprint URI corresponds to the actual public key used is to sign a message containing the correct public key with the private key."

If you have other specific wording improvements in mind, let me know.

File: draft-ietf-oauth-jwk-thumbprint-uri.xml (hunk 3 above)
Comment (selfissued) [reply draft-ietf-oauth-jwk-thumbprint-uri.xml:151]: Agreed

File: draft-ietf-oauth-jwk-thumbprint-uri.xml (hunk 1 above)
Comment (Sakurann) [reply draft-ietf-oauth-jwk-thumbprint-uri.xml:141]: right, ok

File: draft-ietf-oauth-jwk-thumbprint-uri.xml (hunk 2 above)
Comment (Sakurann) [reply draft-ietf-oauth-jwk-thumbprint-uri.xml:146]: Interesting. I will think about this..

**Pull request merged**